docs/modules/ROOT/pages/explanation/pod-security-admissions.adoc

```diff
@@ -0,0 +1,103 @@
+
+Kubernetes 1.24 removes the Pod Security Policy feature in favor of Pod Security Admission. This document explains the impact on {product} users.
+
+== Introduction
+
+Up until OpenShift 4.10, OpenShift only had Security Context Constraints (SCCs.)
+Since Kubernetes 1.3, Pod Security Policy (PSP) has been the built-in way to do that for security-related Pod fields.
+However, it turned out that PSPs were too complex, so Kubernetes 1.21 https://kubernetes.io/blog/2021/04/08/kubernetes-1-21-release-announcement/#podsecuritypolicy-deprecation[deprecated] and 1.24 completely dropped them in favor of https://kubernetes.io/docs/concepts/security/pod-security-admission/[Pod Security Admission] (PSA.)
+
+PSA is a simpler approach, which is similar–but not identical–to OpenShift's SCCs.
+Pod Security Admission was designed to meet the most common security needs out of the box while being simple to understand and adopt.
+
+== OpenShift 4.11
+
+https://docs.openshift.com/container-platform/4.11/release_notes/ocp-4-11-release-notes.html#ocp-4-11-auth-pod-security-admission[OpenShift 4.11 introduces PSA] globally with restricted audit logging and API warnings, but it doesn't replace SCC with PSA, instead running both mechanisms in parallel.
+
+OpenShift 4.11 already enables PSA in auditing mode (including API warnings) and has additional logic which ensures that SCCs and PSAs don't conflict.
+Since it runs PSA in warning mode, it leads to warnings like the following:
+
+[source]
+--
+Warning: would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "storage-bench" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "storage-bench" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "storage-bench" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "storage-bench" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
+--
+
+At the core of PSA, there are the https://kubernetes.io/docs/concepts/security/pod-security-standards/[Pod Security Standards], which define three different policies to broadly cover the security spectrum.
+These policies are cumulative and range from highly permissive to highly restrictive.
+
+[cols="1,6"]
+|===
+|*Privileged*
+|Unrestricted policy, providing the widest possible level of permissions.
+This policy allows for known privilege escalations.
+
+|*Baseline*
+|Minimally restrictive policy which prevents known privilege escalations.
+Allows the default (minimally specified) Pod configuration.
+
+|*Restricted*
+|Heavily restricted policy, following current Pod hardening best practices.
+|===
+
+The most straightforward way to ensure your workloads can work on a restricted namespace, particularly in 4.12, is by labeling the namespaces where they should run by enforcing the restricted policy and verifying if they're admitted and are successfully running.
```

Comment:
Note: regular APPUiO Cloud users currently don't have sufficient permissions to add the relevant labels:

```
simon@phoenix:~ [appuio-lab] $ kubectl label ns sg-test pod-security.kubernetes.io/restricted=enforce
Error from server: admission webhook "validate.kyverno.svc-fail" denied the request: 

resource Namespace//sg-test was blocked due to the following policies

validate-namespace-metadata:
  validate-labels: |-
    validation failure: The following labels can be modified:
        appuio.io/organization, custom.appuio.io/*, kubernetes.io/metadata.name, network-policies.syn.tools/no-defaults, network-policies.syn.tools/purge-defaults.
    labels given:
        {"appuio.io/organization":"vshn","kubernetes.io/metadata.name":"sg-test","pod-security.kubernetes.io/restricted":"enforce"}.
    labels before modification:
        {"appuio.io/organization":"vshn","kubernetes.io/metadata.name":"sg-test"}.
```

However, it should be fairly straightforward to allow users to set the `pod-security.kubernetes.io/restricted` label on their namespaces.

Reply:
So how can then users make sure that their workloads can work on a restricted namespace?

Reply:
IMO we could just allow them to set the `pod-security.kubernetes.io/restricted` label (this should never enable a user to grant themselves extra privileges, afaict), but maybe @corvus-ch has a different/better idea?

Reply:
Let us, for now, not suggest changing the labels. I created APUB-347 to make that possible. As part of that, we will then update this section again.

Reply:
So we leave this section as it is now?